astacus/coordinator/plugins/base.py

```diff
@@ -575,32 +595,29 @@ class ComputeKeptBackupsStep(Step[Sequence[ipc.BackupManifest]]):
     explicit_delete: Sequence[str]
     retain_deltas: bool = False
 
-    async def run_step(self, cluster: Cluster, context: StepsContext) -> Sequence[ipc.BackupManifest]:
+    async def run_step(self, cluster: Cluster, context: StepsContext) -> Sequence[ManifestMin]:
         kept_manifests = await self.compute_kept_basebackups(context)
         if self.retain_deltas:
             kept_manifests += await self.compute_kept_deltas(kept_manifests, context)
         return kept_manifests
 
-    async def compute_kept_basebackups(self, context: StepsContext) -> List[ipc.BackupManifest]:
+    async def compute_kept_basebackups(self, context: StepsContext) -> List[ManifestMin]:
         all_backup_names = context.get_result(ListBackupsStep)
         kept_backup_names = all_backup_names.difference(set(self.explicit_delete))
         manifests = []
         for backup_name in kept_backup_names:
-            manifests.append(await download_backup_manifest(self.json_storage, backup_name))
-            manifests = _prune_manifests(manifests, self.retention)
```

Comment:
Was this intended to run this repeatedly?  Looks wrong.

Reply:
It was needed for full manifests as we didn't want to download them all and only then start pruning them.